src/EnergyPlus/HeatRecovery.hh

```diff
@@ -94,413 +81,318 @@ namespace HeatRecovery {
+        Real64 T_MinProcAirInRelHum = 0.0;  // min allowable process inlet air relative humidity [%]
+        Real64 T_MaxProcAirInRelHum = 0.0;  // max allowable process inlet air relative humidity [%]
+
+        std::array<Real64, 8> C = {0}; // regeneration outlet humidity ratio equation coefficients and limits
+        // index 0: constant coefficient for outlet regen humidity ratio equation
+        // index 1: regen inlet humidity ratio coefficient for outlet regen humidity ratio eq
+        // index 2: regen inlet temp coefficient for outlet regen humidity ratio equation
+        // index 3: (regen in humrat/regen in temp) coefficient for outlet regen humidity ratio eq
+        // index 4: process inlet humidity ratio coefficient for outlet regen humidity ratio eq
+        // index 5: process inlet temp coefficient for outlet regen humidity ratio eq
+        // index 6: (proc in humrat/proc in temp) coefficient for outlet regen humidity ratio eq
+        // index 7: process, regen face velocity coefficient for outlet regen humidity ratio eq
+
+        Real64 H_MinRegenAirInTemp = 0.0;   // min allowable regen inlet air temperature [C]
+        Real64 H_MaxRegenAirInTemp = 0.0;   // max allowable regen inlet air temperature [C]
+        Real64 H_MinRegenAirInHumRat = 0.0; // min allowable regen inlet air humidity ratio [kg water / kg air]
+        Real64 H_MaxRegenAirInHumRat = 0.0; // max allowable regen inlet air humidity ratio [kg water / kg air]
+        Real64 H_MinProcAirInTemp = 0.0;    // min allowable process inlet air temperature [C]
+        Real64 H_MaxProcAirInTemp = 0.0;    // max allowable process inlet air temperature [C]
+        Real64 H_MinProcAirInHumRat = 0.0;  // min allowable process inlet air humidity ratio [kg water/kg air]
+        Real64 H_MaxProcAirInHumRat = 0.0;  // max allowable process inlet air humidity ratio [kg water/kg air]
+        Real64 H_MinFaceVel = 0.0;          // min allowable process, regen face velocity [m/s]
+        Real64 H_MaxFaceVel = 0.0;          // max allowable process, regen face velocity [m/s]
+        Real64 MinRegenAirOutHumRat = 0.0;  // min allowable regen outlet air temperature [C]
+        Real64 MaxRegenAirOutHumRat = 0.0;  // max allowable regen outlet air temperature [C]
+        Real64 H_MinRegenAirInRelHum = 0.0; // min allowable regen inlet air relative humidity [%]
+        Real64 H_MaxRegenAirInRelHum = 0.0; // max allowable regen inlet air relative humidity [%]
+        Real64 H_MinProcAirInRelHum = 0.0;  // min allowable process inlet air relative humidity [%]
+        Real64 H_MaxProcAirInRelHum = 0.0;  // max allowable process inlet air relative humidity [%]
 
-        // Default Constructor
-        HeatExchCondNumericFieldData()
-        {
-        }
-    };
+        // for model bound checking
+        ErrorTracker regenInRelHumTempErr;    // regen inlet relative humidity for temperature equation
```

Comment:
That huge list of bools, ints, strings, and floats, are replaced with meaningful ErrorTracker class instances.  It is waaay easier to track the individual errors now.  (For the record, I think this is an insane number of errors to track, but this PR is not doing functional changes, just refactoring existing code, so that could be discussed later...)